src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationFeedFormatter.cs

```diff
@@ -2,191 +2,169 @@
 
         internal static protected SyndicationCategory CreateCategory(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationCategory>(feed.CreateCategory(), SR.FeedCreatedNullCategory);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreateCategory(), SR.FeedCreatedNullCategory);
         }
 
         internal static protected SyndicationCategory CreateCategory(SyndicationItem item)
         {
             if (item == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("item");
+                throw new ArgumentNullException(nameof(item));
             }
-            return GetNonNullValue<SyndicationCategory>(item.CreateCategory(), SR.ItemCreatedNullCategory);
+
+            return GetNonNullValue(item.CreateCategory(), SR.ItemCreatedNullCategory);
         }
 
         internal static protected SyndicationItem CreateItem(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationItem>(feed.CreateItem(), SR.FeedCreatedNullItem);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreateItem(), SR.FeedCreatedNullItem);
         }
 
         internal static protected SyndicationLink CreateLink(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationLink>(feed.CreateLink(), SR.FeedCreatedNullPerson);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreateLink(), SR.FeedCreatedNullPerson);
         }
 
         internal static protected SyndicationLink CreateLink(SyndicationItem item)
         {
             if (item == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("item");
+                throw new ArgumentNullException(nameof(item));
             }
-            return GetNonNullValue<SyndicationLink>(item.CreateLink(), SR.ItemCreatedNullPerson);
+
+            return GetNonNullValue(item.CreateLink(), SR.ItemCreatedNullPerson);
         }
 
         internal static protected SyndicationPerson CreatePerson(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationPerson>(feed.CreatePerson(), SR.FeedCreatedNullPerson);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreatePerson(), SR.FeedCreatedNullPerson);
```

Comment:
I really don't want to go through every CreateXXXXX method in this class and manually find a call path which can pass a null. As every one I've checked so far has been able to receive a null, I am confident the best option is to revert all these Debug.Assert changes for the CreateXXXXX methods back to throwing an exception.